src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1084,30 +1095,27 @@ def dbapi(
                     )
                     for i, query in enumerate(partitioned_queries)
                 ]
-
-                completed_futures = wait(futures, return_when=ALL_COMPLETED)
-            files = []
-            for f in completed_futures.done:
-                if isinstance(f.result(), Exception):
-                    raise f.result()
-                else:
-                    files.append(f.result())
-            with ThreadPoolExecutor(max_workers=max_workers) as thread_executor:
-                futures = [
-                    thread_executor.submit(
-                        self.upload_and_copy_into_table,
-                        f,
-                        snowflake_stage_name,
-                        snowflake_table_name,
-                        "abort_statement",
-                    )
-                    for f in files
-                ]
-
-                completed_futures = wait(futures, return_when=ALL_COMPLETED)
-            for f in completed_futures.done:
-                if f.result() is not None:
-                    raise f.result()
```

Comment:
it just came to my mind that timeout is an important parameter which we're missing: **query_timeout**, the max timeout used to fetch source data.

we probably need another timeout to control our ingestion process.

can you log a JIRA for timeout support and mark it as HIGH?